stdlib/public/core/Map.swift

```diff
@@ -81,7 +81,10 @@ public struct LazyMapCollection<Base : Collection, Element>
 
   public var startIndex: Base.Index { return _base.startIndex }
```

Comment:
I think that we would need to gyb three variants of LazyMapCollection now, for forward, bidirectional and random access traversal, so that the mapped collection preserves the traversal of the base collection.  Please add a fixme.